### PR TITLE
Main fix test error + upd

### PR DESCRIPTION
**Pull request by SurfaceS**
fix error :

Seems that tmdb has no apikey set -> result null -> omdb fetch.
I put mocks on tmdb and code related to it (key provided = real fetch, no key or 'foo' key = mocks).

Omdb data seems to give a wrong series IMDbID (give episode IMDbID = series IMDbID).
I put some code to get series IMDbID from tmdb to temp fix that.
Updated the mocks with value given by the omdb server (with errored series IMDbID).

update yarn.
update deps.

### CODE REVIEW

**Comment by SubJunk** [comment]
@SurfaceS `TMDB_API_KEY` is set in GitHub so I'm not sure why that wouldn't be working:

image

image

**Comment by SurfaceS** [comment]
on my code :
```
const apiKey = process.env.TMDB_API_KEY || 'foo';
const baseUrl = apiKey === 'foo' ? 'https://local.themoviedb.org/3/' : null;
```
So if it take the mocks url (I name it local.themoviedb.org), it mean TMDB_API_KEY is null or 'foo'.

File: test/e2e/media-video.spec.ts [hunk 1]
```diff
@@ -280,6 +283,7 @@ describe('get by all', () => {
       openSubsSpy.mockReset();
 
       // subsequent calls should return MongoDB result rather than calling external apis
+      // FIXME : try to fetch OMDb with seriesIMDbID as IMDb ID.
```
Comment (SubJunk) [inline test/e2e/media-video.spec.ts:286]: Is this meant for now or later?

**Comment by SurfaceS** [comment]
**Concerning the PR :**
PR is ok for me.

The error in the check `Series or season not found!` is normal as omdb doesn't know the season 1 for 'Lost'.
OMDB seems to be a way less accurate than TMDB.

**Concerning the api :**

The api is indexed on imdb id's, but sometimes it is only tmdb id specially for series.
The api is english only request/reply.

I'll back on another PR about that.

**Comment by SubJunk** [comment]
We can probably just remove OMDb. Unfortunately we found it before TMDB. If we had found TMDB first, we probably wouldn't have added OMDb.
And yes I agree it would be good to fix for entries without IMDb IDs.

**Review by SubJunk — APPROVED** [review]

**Comment by SubJunk** [comment]
Nice fixes thanks!

**Pull request merged**